uniswap-v2/src/store_key.rs

```diff
@@ -0,0 +1,144 @@
+use substreams::scalar::BigInt;
+
+#[derive(Clone)]
+pub(crate) enum StoreKey {
+    Pool,
+    TotalBalance,
+    LatestTimestamp,
+    LatestBlockNumber,
+    TokenBalance,
+    Token0Balance,
+    Token1Balance,
+    OutputTokenBalance,
+    TokenPrice,
+    TotalValueLockedUSD,
+    Volume,
+    DailySupplySideRevenueUSD,
+    HourlySupplySideRevenueUSD,
+    CumulativeSupplySideRevenueUSD,
+    DailyProtocolSideRevenueUSD,
+    HourlyProtocolSideRevenueUSD,
+    CumulativeProtocolSideRevenueUSD,
+    DailyTotalRevenueUSD,
+    HourlyTotalRevenueUSD,
+    CumulativeTotalRevenueUSD,
+    DailyVolumeUSD,
+    HourlyVolumeUSD,
+    CumulativeVolumeUSD,
+    VolumeByTokenUSD,
+    DailyVolumeByTokenUSD,
+    HourlyVolumeByTokenUSD,
+    DailyVolumeByTokenAmount,
+    HourlyVolumeByTokenAmount,
+}
+
+impl StoreKey {
+    pub fn get_unique_pool_key(&self, key: &String) -> String {
+        format!("{}:{}", self.unique_id(), key)
+    }
+
+    pub fn get_unique_pair_key(&self, key1: &String, key2: &String) -> String {
+        format!("{}:{}:{}", self.unique_id(), key1, key2)
+    }
+
+    pub fn get_unique_protocol_key(&self) -> String {
+        format!("[Protocol]:{}", self.unique_id())
+    }
+
+    pub fn get_unique_daily_pool_key(&self, day_id: BigInt, key: &String) -> String {
```

Comment:
Accept `&BigInt` arg instead and then all uses of fn should take refs instead. Same for other fn below